usaspending_api/references/v2/views/toptier_agencies.py

```diff
@@ -59,73 +50,148 @@ def get(self, request, format=None):
+            }
+            for val in latest_submission_by_toptier
+        }
+
+        # Add a placeholder for Agencies that have submitted but not in recent Quarter / Period;
+        # Placeholder is most recent Period window to ensure accurate "current_total_budget_authority_amount"
+        placeholder = {
+            "fiscal_year": most_recent_period_window["submission_fiscal_year"],
+            "fiscal_quarter": most_recent_period_window["submission_fiscal_quarter"],
+            "fiscal_period": most_recent_period_window["submission_fiscal_month"],
+        }
+        for toptier_code in all_toptier_with_submission:
+            submission_values = latest_submission_by_toptier.get(toptier_code)
+            if submission_values is None:
+                latest_submission_by_toptier[toptier_code] = placeholder
+
+        aab_sums_by_toptier = (
+            AppropriationAccountBalances.objects.filter(submission__is_final_balances_for_fy=True)
+            .values(
+                "treasury_account_identifier__funding_toptier_agency",
+                "submission__reporting_fiscal_year",
+                "submission__reporting_fiscal_quarter",
+            )
+            .annotate(
                 budget_authority_amount=Coalesce(Sum("total_budgetary_resources_amount_cpe"), 0),
                 obligated_amount=Coalesce(Sum("obligations_incurred_total_by_tas_cpe"), 0),
                 outlay_amount=Coalesce(Sum("gross_outlay_amount_by_tas_cpe"), 0),
             )
+        )
+        aab_sums_by_toptier = {
+            (
+                val["treasury_account_identifier__funding_toptier_agency"],
+                val["submission__reporting_fiscal_year"],
+                val["submission__reporting_fiscal_quarter"],
+            ): {
+                "budget_authority_amount": val["budget_authority_amount"],
+                "obligated_amount": val["obligated_amount"],
+                "outlay_amount": val["outlay_amount"],
+            }
+            for val in aab_sums_by_toptier
+        }
 
-            abbreviation = ""
-            if toptier_agency.abbreviation is not None:
-                abbreviation = toptier_agency.abbreviation
+        # get agency queryset, distinct toptier id to avoid duplicates, take first ordered agency id for consistency
+        agency_list = (
+            Agency.objects.order_by("toptier_agency_id", "id")
+            .distinct("toptier_agency_id")
+            .values()
+            .annotate(
+                justification=F("toptier_agency__justification"),
+                toptier_abbreviation=F("toptier_agency__abbreviation"),
+                toptier_name=F("toptier_agency__name"),
+                toptier_code=F("toptier_agency__toptier_code"),
+            )
+        )
+
+        for agency in agency_list:
+            submission = latest_submission_by_toptier.get(agency["toptier_code"])
+
+            if submission is None:
```

Comment:
The placeholder is added for Agencies that have a valid submission, but are not in the most recent submission FY and FQ / FP combination. This `None` check is to prevent showing an Agency that that does not have a valid submission.

As for the decision to hide that, it is carried forward and also in line with decisions around Agency Stats where we want to show all Agencies that have had a submission.

Reply:
👍